api/app/references/tests.py

```diff
@@ -94,3 +95,66 @@ def test_reference_api(self):
         reference_request = reference_request_repository.get_by_id(1)
         self.assertEqual(reference_request.has_reference(), True)
 
+    def test_reference_api_update(self):
+        self._seed_data()
+        reference_req = ReferenceRequest(1, 'Mr', 'John', 'Snow', 'Supervisor', 'common@email.com')
+        reference_request_repository.create(reference_req)
+        REFERENCE_DETAIL = {
+            'token':reference_req.token,
+            'uploaded_document': 'DOCT-UPLOAD-78999',
+        }
+        REFERENCE_DETAIL_2 = {
+            'token':reference_req.token,
+            'uploaded_document': 'DOCT-UPLOAD-79000',
+        }
+        response = self.app.post(
+            '/api/v1/reference', data=REFERENCE_DETAIL, headers=self.headers)
+        self.assertEqual(response.status_code, 201)
+
+        response = self.app.put(
+            '/api/v1/reference', data=REFERENCE_DETAIL_2, headers=self.headers)
+        self.assertEqual(response.status_code, 200)
+
+        response = self.app.get(
+            '/api/v1/reference', data={'response_id':1}, headers=self.headers)
+        LOGGER.debug(response.data)
+        data = json.loads(response.data)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(data), 1)
+        reference_request = reference_request_repository.get_by_id(1)
+        self.assertEqual(reference_request.has_reference(), True)
```

Comment:
Can we perhaps do an assert to see if the `uploaded_document` field actually changed to `DOCT-UPLOAD-79000`?